### PR TITLE
feat: add nodeSelector and tolerations support

### PR DESCRIPTION
**Pull request by mbelang**
Allows to deploy on a specific set of nodes.

### CODE REVIEW

**Comment by mbelang** [comment]
I deployed that chart on my development cluster and it scheduled on the node I wanted to. I do not know if I need to add something in the deploy test here.

File: charts/rasa-x/values.yaml [hunk 1]
```diff
@@ -214,6 +214,13 @@ debugMode: false
 # high-load setups).
 separateEventService: "true"
 
+# Settings for pod assignment
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]: We try to follow the best practices from Helm on documentation the values https://helm.sh/docs/chart_best_practices/values/#document-valuesyaml .
```suggestion
# tolerations can be used to control the pod to node assignment
# https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
```

File: charts/rasa-x/values.yaml [hunk 2]
```diff
@@ -214,6 +214,13 @@ debugMode: false
...
+tolerations: []
+  # - key: "nvidia.com/gpu"
+  #   operator: "Exists"
+nodeSelector: {}
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]:
```suggestion
# nodeSelector to specify which node the pods should run on
# https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
nodeSelector: {}
```

**Comment by wochinge** [comment]
@mbelang Sorry, I merged my other PR so you'd have to rebase please.

**Comment by mbelang** [comment]
> rasa containers on a machine with a GPU, but rasa-x or duckling don't require that and there is no benefit. What do you think?

You are totally right. my usecase is no GPU but this is something that someone would want. I'll do the changes.

**Comment by mbelang** [comment]
@wochinge all comments addressed. Tested, working fine.

**Comment by wochinge** [comment]
@mbelang I think the `lint` part will fail because you have to add the `tolerations` / `nodeSelector` key to every part in the `values.yml`

**Comment by mbelang** [comment]
Ci is passing now no?

**Comment by wochinge** [comment]
> Ci is passing now no?

Yep, however I think we should still add the values to every component in `values.yml`

**Comment by mbelang** [comment]
what do you mean? I thought we said only rasa component have it's own values. But if you want all of them to bit split, I can do it too.

**Comment by mbelang** [comment]
@wochinge I'm sorry I read your comment again and I do not understand it.

File: charts/rasa-x/values.yaml [hunk 3]
```diff
@@ -216,6 +228,17 @@ debugMode: false
...
+# https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+tolerations: []
+  # - key: "nvidia.com/gpu"
+  #   operator: "Exists"
+
+# nodeSelector to specify which node the pods should run on
+# https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+nodeSelector: {}
+  # "beta.kubernetes.io/instance-type": "g3.8xlarge"
+
```
[the remark this answers is not in the document]
Comment (wochinge) [reply charts/rasa-x/values.yaml]: @mbelang

File: charts/rasa-x/values.yaml [hunk 4]
```diff
@@ -88,6 +88,18 @@ rasa:
     #     secretKeyRef:
     #       name: twilio-auth
     #       key: twilio_auth_token
+
+  # tolerations can be used to control the pod to node assignment
```
[the remark this answers is not in the document]
Comment (wochinge) [reply charts/rasa-x/values.yaml]: @mbelang I would copy and past these section to `rasax`, `app`, and so in `values.yml` (and not justing having it for `rasa`

[the remark this answers is not in the document]
Comment (mbelang) [reply charts/rasa-x/values.yaml]: Ok done :)

**Comment by mbelang** [comment]
All done @wochinge

**Comment by mbelang** [comment]
Is the master branch pipeline faillure normal?

**Comment by wochinge** [comment]
> Is the master branch pipeline faillure normal?

Can you send a link? Everything here seems fine https://github.com/RasaHQ/rasa-x-helm/actions

**Comment by mbelang** [comment]
https://github.com/mbelang/rasa-x-helm/runs/519265498

**Comment by wochinge** [comment]
This is because GitHub doesn't pass the secrets of this repository to builds which are running in your repository.

**Comment by mbelang** [comment]
oh man :man_facepalming: this is my fork master branch. nevermind, I'm drunk lol.

**Pull request merged**